### PR TITLE
Knative local

### PR DESCRIPTION
**Pull request by seabaylea**
##### Checklist
- [X] commit message follows [commit guidelines](https://github.com/appsody/docs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

This adds documentation on how to set up a local Knative install using Kubernetes in Docker for Desktop and links to it from the deploying guide.

### CODE REVIEW

File: docs/using-appsody/installing-knative-locally.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+Select the Docker icon in the Menu bar, click **Preferences** and select the **Kubernetes** tab. Here select the **Enable Kubernetes** checkbox and click **Apply**.
+
+Click **Install** on the dialog box asking, “Install the Kubernetes Cluster now?” to start the install. When it’s complete it will have installed Kubernetes along with the `kubectl` command line tool which `appsody` will use to deploy your applications.
+
+Next ensure that Kubernetes has enough resources to run your apps by selecting the Docker icon in the Menu bar, click **Preferences** and select the **Advanced** tab. Use the sliders to ensure that you have 6 CPUs and 8.0 GiB of memory assigned to Kubernetes and click **Apply & Restart**.
+
+You can now use the following commands to install Knative Serving 0.6.0 (currently the latest version) and its Istio dependency using the following commands:
+
+1. Install Istio 0.5.2:
```
Comment (neeraj-laad) [inline docs/using-appsody/installing-knative-locally.md]: I don't think this is Istio 0.5.2, this is Istio that comes with knative 0.5.2

File: docs/using-appsody/installing-knative-locally.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	
+2. Verify the STATUS of the Istio components:
+
+	```sh
+	kubectl get pods --namespace istio-system --watch
+	```
+	
+	Once all of the components have reached `Running` or `COMPLETED` you can stop watching the status using `Ctrl-C`.
+	
+3. Install Knative Serving 0.6.0:
```
Comment (neeraj-laad) [inline docs/using-appsody/installing-knative-locally.md]: Slightly odd that we are using 0.6.0 for serving and 0.5.2 for Istio.. They might not have changed Istio but might be cleaner to use a consistent version. And perhaps use the latest 0.7 version?

Comment (seabaylea) [reply docs/using-appsody/installing-knative-locally.md]: This is based on the install instructions on https://knative.dev/docs/install/knative-with-docker-for-mac/ which is using different versions of Knative and Istio - its also what I've tested with and know works ;-P

File: docs/using-appsody/installing-knative-locally.md (hunk 1 above)
Comment (seabaylea) [reply docs/using-appsody/installing-knative-locally.md]: Ok - I'll remove the reference to the version.

File: docs/using-appsody/installing-knative-locally.md (hunk 2 above)
Comment (seabaylea) [reply docs/using-appsody/installing-knative-locally.md]: I've just checked - 0.5.2 was the latest Knative release where they published an istio.yaml

**Comment by seabaylea** [comment]
@neeraj-laad Is this ok as-is? I've tested Knative 0.6.0 - I'll uplift the docs to 0.7.x once I've tested that it still works without breaking anything in `appsody deploy`.

**Comment by seabaylea** [comment]
> I am not convinced that having these instructions in our documentation is a great idea. This will soon get out of sync with official documentation on Knative install process.

The problem (at the moment), is that the Knative serving install instructions from Knative are wrong - if you follow them you'll be unable to deploy an app that it reachable through the browser.

Hopefully we can get them to fix their instructions, and then we can remove these.

File: docs/using-appsody/installing-knative-locally.md (hunk 2 above)
Comment (neeraj-laad) [reply docs/using-appsody/installing-knative-locally.md]: This seems ok then. I was not aware that they stopped publishing Istio with every release (like they used to)

**Pull request merged**